cmd/nodecmd/upgrade.go

```diff
@@ -0,0 +1,224 @@
+	latestAvagoVersion, err := app.Downloader.GetLatestReleaseVersion(binutils.GetGithubLatestReleaseURL(
+		constants.AvaLabsOrg,
+		constants.AvalancheGoRepoName,
+	))
+	if err != nil {
+		return nil, err
+	}
+	latestSubnetEVMVersion, err := app.Downloader.GetLatestReleaseVersion(binutils.GetGithubLatestReleaseURL(
+		constants.AvaLabsOrg,
+		constants.SubnetEVMRepoName,
+	))
+	if err != nil {
+		return nil, err
+	}
+	rpcVersion, err := vm.GetRPCProtocolVersion(app, models.SubnetEvm, latestSubnetEVMVersion)
+	if err != nil {
+		return nil, err
+	}
+	ansibleNodeIDs, err := ansible.GetAnsibleHostsFromInventory(app.GetAnsibleInventoryDirPath(clusterName))
+	if err != nil {
+		return nil, err
+	}
+	failedNodes := []string{}
+	nodeErrors := []error{}
+	nodesToUpgrade := make(map[string]string)
+	for _, host := range ansibleNodeIDs {
+		if err := app.CreateAnsibleStatusFile(app.GetAvalancheGoJSONFile()); err != nil {
+			failedNodes = append(failedNodes, host)
+			nodeErrors = append(nodeErrors, err)
+			continue
+		}
+		if err := ansible.RunAnsiblePlaybookCheckAvalancheGoVersion(app.GetAnsibleDir(), app.GetAvalancheGoJSONFile(), app.GetAnsibleInventoryDirPath(clusterName), host); err != nil {
+			failedNodes = append(failedNodes, host)
+			nodeErrors = append(nodeErrors, err)
+			continue
+		}
+		vmVersions, err := parseNodeVersionOutput(app.GetAvalancheGoJSONFile())
+		if err != nil {
+			failedNodes = append(failedNodes, host)
+			nodeErrors = append(nodeErrors, err)
+			continue
+		}
+		currentAvalancheGoVersion := vmVersions[constants.PlatformKeyName]
+		var avalancheGoVersionToUpdateTo string
+		var subnetEVMVersionToUpdateTo string
+		for vmName, vmVersion := range vmVersions {
+			// when calling info.getNodeVersion, this is what we get
+			// "vmVersions":{"avm":"v1.10.12","evm":"v0.12.5","n8Anw9kErmgk7KHviddYtecCmziLZTphDwfL1V2DfnFjWZXbE":"v0.5.6","platform":"v1.10.12"}},
+			// we need to get the VM ID of the subnets that the node is currently validating, in the example above it is n8Anw9kErmgk7KHviddYtecCmziLZTphDwfL1V2DfnFjWZXbE
+			if vmName != constants.PlatformKeyName && vmName != constants.EVMKeyName && vmName != constants.AVMKeyName {
+				if vmVersion != latestSubnetEVMVersion {
+					// update subnet EVM version
+					ux.Logger.PrintToUser("Upgrading Subnet EVM version for node %s from version %s to version %s", host, vmVersion, latestSubnetEVMVersion)
+					subnetEVMVersionToUpdateTo = fmt.Sprintf("%s:%s", vmName, latestSubnetEVMVersion)
+				}
+				// find the highest version of avalanche go that is still compatible with current highest rpc
+				avalancheGoVersionToUpdateTo, err = GetLatestAvagoVersionForRPC(rpcVersion)
+				if err != nil {
+					failedNodes = append(failedNodes, host)
+					nodeErrors = append(nodeErrors, err)
+					continue
+				}
+			}
+		}
+		if slices.Contains(failedNodes, host) {
+			continue
+		}
+		if currentAvalancheGoVersion != avalancheGoVersionToUpdateTo {
```

Comment:
can we hace a target version to latest first, and the update if needed by the evm req, and if the current differs the target, do the update